.github/workflows/build-and-test-aarch64-android.yml

```diff
@@ -0,0 +1,38 @@
+name: build and test on android
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+
+jobs:
+  test-in-android-emulator:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    steps:
+      - name: add NDK toolchain to path
+        run: |
+            echo "$ANDROID_NDK_LATEST_HOME/toolchains/llvm/prebuilt/linux-x86_64/bin" >> $GITHUB_PATH
+      - name: git checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - name: cache rust dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: aarch64-android-cargo-and-target-${{ hashFiles('**/Cargo.lock') }}
+      - name: cargo build for aarch64-apple-darwin
+        run:  |
+            rustup target add aarch64-linux-android
```

Comment:
Could we just add this to `rust-toolchain.toml`?

Reply:
> Could we just add this to rust-toolchain.toml?

I think that would make us download this target everywhere. I checked the Android build and it downloads all the targets in `rust-toolchain.toml` which is just slowing us down. We're also running `rustup target add $TARGET` for the QEMU builds so what we're doing here is consistent with the existing workflows.

From the Android CI run with the current `rust-toolchain.toml`:

```
info: syncing channel updates for 'nightly-2024-06-13-x86_64-unknown-linux-gnu'
info: latest update on 2024-06-13, rust version 1.81.0-nightly (8337ba918 2024-06-12)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std' for 'aarch64-unknown-linux-gnu'
info: downloading component 'rust-std' for 'armv7-unknown-linux-gnueabihf'
info: downloading component 'rust-std' for 'i686-unknown-linux-gnu'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
```

What do you think about taking the `targets` out of `rust-toolchain.toml` so we avoid wasting CI bandwidth?

Reply:
Hmm, I guess maybe.  They are useful locally.  I thought they were being cached altogether in CI so that it wouldn't be a big deal.  I didn't realize they were being redownloaded every time.  Could we just cache them?

Reply:
> Could we just cache them?

It is possible but not ideal. We want the caches to contain what we need for reach run, not what any run could need.

Over time, the caches would grow as the host images update their toolchains, as we do the same. The size of each toolchain update would be multiplied by the number of `targets` in `rust-toolchain.toml`.

> Hmm, I guess maybe. They are useful locally. 

I chatted with @rinon and his take was that if you're cross compiling, you should be able to add the toolchains (Rust ones plus the host tools for compiling, assembling, and linking the non-Rust bits). I think that's a fair point. 

What do you think about having a separate PR to document cross-compiling? We could suggest how to add the `targets` to `rust-toolchain.toml` along with the other steps required.

Reply:
decided during offline discussion to comment out `targets`. update cross-compiling instructions in README.